src/app/[locale]/flashcards/learning/[topicId]/page.tsx

```
@@ -53,6 +72,42 @@ export default function Page() {
     } = useFetch<IFlashcardWithReviewPrediction[]>(() => flashcardService.getDueFlashcardsForTopic(topicId));
 
+    useEffect(() => {
+        if (flashcards && flashcards.length > 0) {
+            setInitialTotal(flashcards.length);
+        }
+    }, [flashcards]);
+
+    const chunkSize = Math.max(1, Math.ceil(initialTotal * 0.2)); // luôn >=1
+
+    // Wait for SSE "completed" -> parse -> save local -> redirect to local quiz page
+    useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)